src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -3694,6 +3684,9 @@ namespace winrt::TerminalApp::implementation
         _updateThemeColors();
 
         _updateAllTabCloseButtons();
+
+        // The user may have changed the "show title in titlebar" setting.
+        TitleChanged.raise(*this, nullptr);
```

Comment:
This is the actual heart of the fix. When the settings change, just fire off this event indiscriminately.

Reply:
Yes, it will go through and recalculate the title for every window. That is _so small_ compared to the burden of reloading the entire settings. :)